File: 백준/Bronze/1668. 트로피 진열/트로피 진열.c
#include <stdio.h>

int main(void) { 
   int a;
   scanf("%d", &a);

   int thropy[a], height_left = 0, max;

   for(int i = 0; i < a; i++){
      scanf("%d", &thropy[i]);
   }

   for(int i = 0; i < a; i++){
      if(i == 0){
         height_left += 1;
         max = thropy[i];
      }
      else if(i > 0 && i < a - 1){
         if(thropy[i - 1] < thropy[i] && thropy[i] > max){
            height_left += 1;
            max = thropy[i];
         }
      }
      else{
         if(thropy[i - 1] < thropy[i] && thropy[i] > max){
            height_left += 1;
         }
      }
   }

   int height_right = 0;
   
   for(int i = a - 1; i > -1; i--){
      if(i == a - 1){
         height_right += 1;
         max = thropy[i];
      }
      else if(i > 0 && i < a - 1){
         if(thropy[i + 1] < thropy[i] && thropy[i] > max){
            height_right += 1;
            max = thropy[i];
         }
      }
      else{
         if(thropy[i + 1] < thropy[i] && thropy[i] > max){
            height_right += 1;
         }
      }
   }

   printf("%d\n%d", height_left, height_right);
   
   return 0;
}

File: 백준/Bronze/1668. 트로피 진열/트로피 진열.py
a = int(input())

thropy = []
for _ in range(a):
  ip = int(input())
  thropy.append(ip)

height_left = 1
height_right = 1
max = thropy[0]
for i in range(1, a):
  if thropy[i - 1] < thropy[i] and max < thropy[i]:
    height_left += 1
    max = thropy[i]

max = thropy[a - 1]
for i in range(a - 2, -1, -1):
  if thropy[i + 1] < thropy[i] and max < thropy[i]:
    height_right += 1
    max = thropy[i]

print(height_left)
print(height_right)